### ISSUE TITLE
require-lazy macro may be required with clj/cljs agnostic code

### ISSUE THREAD

**Issue by darwin** [roman01la/uix#40]
I stumbled upon this:
https://github.com/roman01la/uix/blob/27172a3a24e7ee099f6c7e2b137cf2965eb6538c/core/dev/uix/recipes/lazy_loading.cljc#L17-L18

Assuming `require-lazy` is a macro, I believe `[uix.core.alpha :as uix :refer [require-lazy]]` should work in both cases as long as cljs file for uix.core.alpha does `:require-macros [uix.core.alpha]`. I'm not sure about uix.core.alpha implemented as cljc file, but if it was .clj and .cljs it would definitely work.

See "implicit refer" in this blog post: https://blog.fikesfarm.com/posts/2016-03-01-clojurescript-macro-sugar.html

**Comment by darwin** [comment]
Thinking about it more. I would move `require-lazy` into a separate namespace. This way you could ensure additional cljs namespaces on which your macro-emitted code rely. Namely `cljs.loader` and `React`. 

Something like:
```clj
(ns uix.core.lazy-loader
  (:require-macros [uix.core.lazy-loader])
  (:require [cljs.loader]
            [react]))
```

A separate namespace is needed because you don't want to require `cljs.loader` in `uix.core.alpha` for everybody.

**Comment by darwin** [comment]
just FYI, I got it working with shadow-cljs:

[loader.clj](https://github.com/binaryage/cljs-react-three-fiber/blob/master/examples/src/app/react_three_fiber/examples/lib/loader.clj)
[loader.cljs](https://github.com/binaryage/cljs-react-three-fiber/blob/master/examples/src/app/react_three_fiber/examples/lib/loader.cljs)

When used [this way](https://github.com/binaryage/cljs-react-three-fiber/blob/2e4f560466575ec594673abde51679019fea57be/examples/src/app/react_three_fiber/examples/demos.cljs#L2) the cljs file should ensure that all references used in generated code are available.

Btw. using `react/lazy` directly didn't work for me. I had to `(def react-lazy react/lazy)` in my ns and use that "exported" symbol instead. But I think this shortcoming of native/npm requires, with true cljs namespace it would work as expected.

**Comment by roman01la** [comment]
Fixed in fd56e6c0a16
